include/deal.II/matrix_free/shape_info.h

```diff
@@ -313,7 +313,7 @@ namespace internal
        * Constructor that initializes the data fields using the reinit method.
        */
       template <int dim>
-      ShapeInfo(const Quadrature<1> &     quad,
+      ShapeInfo(const Quadrature<dim> &   quad,
```

Comment:
While within an internal namespace, this is still an incompatible change of something that might have appeared in user code, so I would only make the change incompatible if there is a clear benefit for doing so. Do you think it would make sense to add a second template argument `q_dim` here? (I am not entirely sure what is the better approach at this point).